### PR TITLE
feat(array-responses): add new 'array-responses' rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces the new spectral-style
"array-responses" rule, which replaces the legacy
"no_array_response" rule.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [x] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [x] Removed or adjusted testcases (packages/validator/test)
- [x] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [x] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

File: packages/ruleset/src/functions/array-responses.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const { isSdkExcluded } = require('../utils');
+
+module.exports = function(operation, _opts, { path }) {
+  return checkForArrayResponses(operation, path);
+};
+
+function checkForArrayResponses(op, path) {
+  // return [];
```
Comment (pyrooka) [inline packages/ruleset/src/functions/array-responses.js:8]: Is this intentional?

**Review by pyrooka — APPROVED** [review]
Looks good!

**Review by dpopp07 — APPROVED** [review]
Looks good! Left a couple comments for consideration but it looks good as is

File: packages/ruleset/src/functions/array-responses.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * Checks for operations that are defined as returning a top-level array in
+ * a response. This is considered a bad practice.  Instead, the response body
+ * should be defined as an object with a property that contains the array.
+ * This provides flexibility to expand the response body definition in the future.
+ * @param {*} op the operation to check
+ * @param {*} path the array of path segments indicating the "location" of the pathItem within the API definition
+ * @returns an array containing the violations found or [] if no violations
+ */
+function checkForArrayResponses(op, path) {
+  if (!op || isSdkExcluded(op)) {
```
[the remark this answers is not in the document]
Comment (padamstx) [reply packages/ruleset/src/functions/array-responses.js:17]: ok, I'll adjust the logic then.

[the remark this answers is not in the document]
Comment (padamstx) [reply packages/ruleset/src/functions/array-responses.js]: I removed the "excluded" exclusion :)

**Pull request merged**